modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/planner/hints/JoinTypeHintPlannerTest.java

```diff
@@ -405,8 +436,7 @@ public void testDisableJoinTypeInSubquery() throws Exception {
     public void testNestedHintOverrides() throws Exception {
         assertPlan("SELECT /*+ " + MERGE_JOIN + "(TBL1) */ t1.v1, t2.v2 FROM TBL1 t1 JOIN TBL2 t2 on t1.v3=t2.v3 " +
                 "where t2.v1 in (SELECT t3.v3 from TBL3 t3 JOIN TBL1 t4 on t3.v2=t4.v2)", schema,
-            nodeOrAnyChild(isInstanceOf(IgniteNestedLoopJoin.class)).negate()
-                .and(nodeOrAnyChild(isInstanceOf(IgniteMergeJoin.class))), CORE_JOIN_REORDER_RULES);
+            nodeOrAnyChild(isInstanceOf(IgniteMergeJoin.class).negate().and(hasNestedTableScan("TBL1"))), CORE_JOIN_REORDER_RULES);
```

Comment:
Why plan is changed here? As far as I can understand there still should be MERGE_JOIN for TBL1 and TBL2.

Reply:
Calcite adds global `IgniteNestedLoopJoin` or `IgniteCorrelatedNestedLoopJoin`. I've made a fix for better processing joins without table scan inputs.